src/mito2/src/manifest/storage.rs

```diff
@@ -405,9 +417,42 @@ impl ManifestObjectStore {
         let path = self.checkpoint_file_path(version);
         // Due to backward compatibility, it is possible that the user's checkpoint not compressed,
         // so if we don't find file by compressed type. fall back to checkpoint not compressed find again.
-        let checkpoint_data =
-            match self.object_store.read(&path).await {
-                Ok(checkpoint) => {
+
+        // get the stored_checksum, which is in the checkpoint metadata.
+        let metadata_path = self.last_checkpoint_path();
+        let metadata_bytes = match self.object_store.read(&metadata_path).await {
+            Ok(metadata_bytes) => metadata_bytes,
+            Err(e) if e.kind() == ErrorKind::NotFound => return Ok(None), // No metadata found
+            Err(e) => return Err(e).context(OpenDalSnafu),                // Error reading metadata
+        };
```

Comment:
Consider providing a helper method (e.g. `read_bytes()`) to read bytes from the object store and return `Ok(None)` if the object is not found.
```suggestion
        let Some(metadata_bytes) = self.read_bytes(&metadata_path).await? else {
            return Ok(None); // No metadata found
        };
```

We could use this method in other places below.